### PR TITLE
Feature: min and max

### PR DESCRIPTION
**Pull request by brennanjl**
This PR adds the min and max aggregate functions. We have had a few people have problems not having this, so I felt it was appropriate to add. It required a non-trivial change to how our lexer and parser work, since `max` and `min` were keywords, and were therefore being lexed and not recognized as function names.

### CODE REVIEW

**Comment by brennanjl** [comment]
This PR is based on https://github.com/kwilteam/kwil-db/pull/777

**Review by jchappelow — DISMISSED** [review]
The `[added min and max](https://github.com/kwilteam/kwil-db/pull/779/commits/ad702ea4f9736d254a560f03516485f2fe62f6d7)` commit looks fine to me, but @Yaiba would have more to say.

What's the `primary_key` vs `primary key` change?  Is that something to note in https://github.com/kwilteam/kwil-db/discussions/668?

**Comment by brennanjl** [comment]
> What's the primary_key vs primary key change?

To specify a primary key, we support:
- `primary_key`
- `pk`
- `primary`
- `primary key`

I simply changed the parser unit tests to test all of these.

File: parse/grammar/KuneiformParser.g4 [hunk 1]
```diff
@@ -127,14 +127,8 @@ typed_variable_list:
...
-    MIN LPAREN literal RPAREN           # min_constraint
-    | MAX LPAREN literal RPAREN         # max_constraint
-    | MINLEN LPAREN literal RPAREN      # min_len_constraint
-    | MAXLEN LPAREN literal RPAREN      # max_len_constraint
-    | (NOTNULL|NOT NULL)                # not_null_constraint
-    | (LEGACY_PRIMARY_KEY|PRIMARY KEY?) # primary_key_constraint
-    | DEFAULT LPAREN literal RPAREN     # default_constraint
-    | UNIQUE                            # unique_constraint
+    // conditionally allow some tokens, since they are used elsewhere
+    (IDENTIFIER| PRIMARY KEY? | NOT NULL | DEFAULT | UNIQUE) (LPAREN literal RPAREN)?
```
Comment (Yaiba) [inline parse/grammar/KuneiformParser.g4:131]: seems no need to declare `DEFAULT` & `UNIQUE` here, in this rule we only need either a single word `IDENTIFIER` or multi word keywords.

**Review by Yaiba — APPROVED** [review]
LGTM

File: parse/grammar/KuneiformParser.g4 (hunk 1 above)
Comment (brennanjl) [reply parse/grammar/KuneiformParser.g4:131]: UNIQUE and DEFAULT are both tokens used else where. DEFAULT is used in foreign keys: `on update set default`, and unique is used in indexes: `#idx unique(column1, column2)`

Comment (Yaiba) [reply parse/grammar/KuneiformParser.g4:131]: ah I just mean for `constrains` rule, since we check the 'syntax' later, but nbd

**Review by Yaiba — APPROVED** [review]

**Pull request merged**